File: 1-TwoSum/1-TwoSum.java
// Last updated: 9/3/2025, 9:24:35 PM
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int j = 0;
        int i = 1;
        int c = 0;
        while (j < nums.length) {
            if (j == i) {
                i++;
                continue;
            } else if ((nums[j] + nums[i]) != target) {
                i++;
                if (i == nums.length) {
                   
                    i = 0;
                    j++;
                }
            } else if ((nums[j] + nums[i]) == target) {
                return new int[] { j, i };
            }
            
        }
        return new int[] {}; 
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 9/3/2025, 9:24:36 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int j = 0;
        int i = 1;
        int c = 0;
        while (j < nums.size()) {
            if (j == i) {
                i++;
                continue;
            }
            else if ((nums[j] + nums[i]) != target) {
                i++;
                if (i == nums.size()) {
                    c = 0;
                    i = 0;
                    j++;
                }
            }
            else if ((nums[j] + nums[i]) == target) {
                return { j,i };
            }
            c++;
        }
        return {}; 
}
};
